backends/pkg/reconciler/eventmesh/handler.go

```diff
@@ -47,14 +43,20 @@ type EventMesh struct {
 const BackstageKubernetesIDLabel = "backstage.io/kubernetes-id"
 
 // HttpHandler is the HTTP handler that's used to serve the event mesh data.
-func HttpHandler(ctx context.Context, listers Listers) func(w http.ResponseWriter, req *http.Request) {
+func HttpHandler(ctx context.Context, config *rest.Config) func(w http.ResponseWriter, req *http.Request) {
 	logger := logging.FromContext(ctx)
 
 	// this handler simply calls the event mesh builder and returns the result as JSON
 	return func(w http.ResponseWriter, req *http.Request) {
 		logger.Debugw("Handling request", "method", req.Method, "url", req.URL)
 
-		eventMesh, err := BuildEventMesh(ctx, listers, logger)
+		config.BearerToken = req.Header.Get("Authorization")
```

Comment:
This code will update the config that's passed to the variable. That will mean the token will be saved in memory.

I suggest that we recreate a config based on the incluster config (to use Kube API server, certs, etc) using
```
rest.CopyConfig(config)
```

I would even pass a copy of `config` from `controller.go` like

```go
...
	config, err := rest.InClusterConfig()
	if err != nil {
		log.Fatalf("Error getting in-cluster config: %v", err)
	}
	
	noTokenConfig := rest.CopyConfig(config)
	noTokenConfig.BearerToken = ""
	noTokenConfig.Username = ""
	noTokenConfig.Password = ""
	noTokenConfig.BearerTokenFile = ""

	r.HandleFunc("/", HttpHandler(ctx, noTokenConfig)).Methods("GET")
	http.Handle("/", r)
...
```

And, create another copy in this function like:

```go
func HttpHandler(ctx context.Context, inClusterConfig *rest.Config) func(w http.ResponseWriter, req *http.Request) {
	logger := logging.FromContext(ctx)

	// this handler simply calls the event mesh builder and returns the result as JSON
	return func(w http.ResponseWriter, req *http.Request) {
		logger.Debugw("Handling request", "method", req.Method, "url", req.URL)

		config := rest.CopyConfig(inClusterConfig)
		config.BearerToken = req.Header.Get("Authorization")
		
		clientset, err := versioned.NewForConfig(config)
...
```

Reply:
Thanks for wonderful comments @aliok ! i am working on it.

Reply:
I made changes but haven't made a detailed test yet , regards.

Reply:
I haven't tested that code myself too, when I was making the suggestion. But, I have faith that my suggestion will work 😆

Reply:
<img width="614" alt="Screenshot 2024-06-07 at 03 27 58" src="https://github.com/knative-extensions/backstage-plugins/assets/37239627/a9876420-c5ee-4d58-9461-1dfef0c18bb9">
<img width="1024" alt="Screenshot 2024-06-07 at 03 28 05" src="https://github.com/knative-extensions/backstage-plugins/assets/37239627/773d0d1e-5f78-4938-9043-2ac0bcd1760f">

Yes, I tried the steps in  below and it is working. Thank you again.